### ISSUE TITLE
Feature request: support for native Ecma script modules

### ISSUE THREAD

**Issue by Havunen** [Aymkdn/html-to-pdfmake#192]
I'm requesting this module to add support for native ESM modules so it can be loaded in environments where common js is not supported. For example jest native ESM mode.

add `"type": "module"` to package.json and "exports" entry point

**Comment by Aymkdn** [comment]
I'm afraid that it will break the module import for many people (including me)… depending on the version of Node and how old your projects are.

I'd have to run some tests, but it's not on my priority list right now, so don't hold your breath!

**Comment by Havunen** [comment]
What about bumping major version to v3 to indicate the braking change and supporting both v2 and v3 temporarily?

**Comment by Aymkdn** [comment]
> What about bumping major version to v3 to indicate the braking change and supporting both v2 and v3 temporarily?

Yes, but it means more work for me because I'll have to maintain two branches. I'll think about it.

**Comment by SatishB15** [comment]
Solution for using html-to-pdfmake in module-based environments (e.g., Angular 18, Jest ESM)

I encountered the same issue while using html-to-pdfmake in Angular 18 due to its CommonJS implementation not being compatible with native ES Modules. Instead of the standard:

```js
import * as htmlToPdfMake from 'html-to-pdfmake';
```

I used the following alternative syntax:


```js
import _ from 'html-to-pdfmake';
```

This syntax resolved the issue and allowed the library to function seamlessly in an ESM environment.

Why it Works
	•	This syntax ensures compatibility with CommonJS modules in environments expecting ES Module exports.
	•	The underscore _ correctly handles the default export from the CommonJS module.

**Comment by guilhermebentomarques** [comment]
To me using Angular 19 it worked like this

```
import pdfMake from 'pdfmake/build/pdfmake';
import pdfFonts from 'pdfmake/build/vfs_fonts';
import htmlToPdfmake from 'html-to-pdfmake';

pdfMake.vfs = pdfFonts.vfs;
```

Versions

```
"html-to-pdfmake": "^2.5.21",
"pdfmake": "^0.2.18",
"@types/html-to-pdfmake": "^2.4.4",
"@types/pdfmake": "^0.2.11",
```
